src/joystick/hidapi/SDL_hidapi_switch2.c

```diff
@@ -0,0 +1,266 @@
+{
+    SDL_AddHintCallback(SDL_HINT_JOYSTICK_HIDAPI_SWITCH2, callback, userdata);
+}
+
+static void HIDAPI_DriverSwitch2_UnregisterHints(SDL_HintCallback callback, void *userdata)
+{
+    SDL_RemoveHintCallback(SDL_HINT_JOYSTICK_HIDAPI_SWITCH2, callback, userdata);
+}
+
+static bool HIDAPI_DriverSwitch2_IsEnabled(void)
+{
+    return SDL_GetHintBoolean(SDL_HINT_JOYSTICK_HIDAPI_SWITCH2, SDL_GetHintBoolean(SDL_HINT_JOYSTICK_HIDAPI, SDL_HIDAPI_DEFAULT));
+}
+
+static bool HIDAPI_DriverSwitch2_IsSupportedDevice(SDL_HIDAPI_Device *device, const char *name, SDL_GamepadType type, Uint16 vendor_id, Uint16 product_id, Uint16 version, int interface_number, int interface_class, int interface_subclass, int interface_protocol)
+{
+    if (vendor_id == USB_VENDOR_NINTENDO) {
+        if (product_id == USB_PRODUCT_NINTENDO_SWITCH2_GAMECUBE_CONTROLLER) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static bool HIDAPI_DriverSwitch2_InitDevice(SDL_HIDAPI_Device *device)
+{
+    return HIDAPI_JoystickConnected(device, NULL);
+}
+
+static int HIDAPI_DriverSwitch2_GetDevicePlayerIndex(SDL_HIDAPI_Device *device, SDL_JoystickID instance_id)
+{
+    return -1;
+}
+
+static void HIDAPI_DriverSwitch2_SetDevicePlayerIndex(SDL_HIDAPI_Device *device, SDL_JoystickID instance_id, int player_index)
+{
+}
+
+static bool HIDAPI_DriverSwitch2_UpdateDevice(SDL_HIDAPI_Device *device)
+{
+    const struct {
+        int byte;
+        unsigned char mask;
+    } buttons[] = {
+        {3, 0x01}, // B
+        {3, 0x02}, // A
+        {3, 0x04}, // Y
+        {3, 0x08}, // X
+        {3, 0x10}, // R (GameCube R Click)
+        {3, 0x20}, // ZR (GameCube Z)
+        {3, 0x40}, // PLUS (GameCube Start)
+        {3, 0x80}, // RS (not on GameCube)
+        {4, 0x01}, // DPAD_DOWN
+        {4, 0x02}, // DPAD_RIGHT
+        {4, 0x04}, // DPAD_LEFT
+        {4, 0x08}, // DPAD_UP
+        {4, 0x10}, // L (GameCube L Click)
+        {4, 0x20}, // ZL
+        {4, 0x40}, // MINUS (not on GameCube)
+        {4, 0x80}, // LS (not on GameCube)
+        {5, 0x01}, // Home
+        {5, 0x02}, // Capture
+        {5, 0x04}, // GR (not on GameCube)
+        {5, 0x08}, // GL (not on GameCube)
+        {5, 0x10}, // C
+    };
+
+    SDL_Joystick *joystick = NULL;
+    if (device->num_joysticks > 0) {
+        joystick = SDL_GetJoystickFromID(device->joysticks[0]);
+    }
+    if (joystick == NULL) {
+        return true;
+    }
+
+    // Read input packet
+
+    Uint8 packet[USB_PACKET_LENGTH];
+    int size;
+    while ((size = SDL_hid_read_timeout(device->dev, packet, sizeof(packet), 0)) > 0) {
+        if (size < 15) {
+            continue;
+        }
+
+        Uint64 timestamp = SDL_GetTicksNS();
+        for (size_t i = 0; i < SDL_arraysize(buttons); ++i) {
+            SDL_SendJoystickButton(
+                timestamp,
+                joystick,
+                (Uint8) i,
+                (packet[buttons[i].byte] & buttons[i].mask) != 0
```

Comment:
After the latest change to the `buttons` array, it seems fairly redundant from a purely functional standpoint.

```suggestion
                (packet[3 + (i >> 3)] & (1U << (i & 7))) != 0
```